app/services/assessors/vehicle_assessor.rb

```diff
@@ -1,9 +1,55 @@
 module Assessors
   class VehicleAssessor
+    Result = Struct.new(:value, :included_in_assessment)
     class << self
-      def call(vehicles)
-        vehicles.each(&:assess!)
-        vehicles.sum(&:assessed_value)
+      def call(vehicles, submission_date)
+        vehicles.sum { assess(_1, submission_date) }
+      end
+
+      def assess(vehicle, submission_date)
+        result = if vehicle.in_regular_use?
+                   assess_in_regular_use(vehicle, submission_date)
+                 else
+                   assess_not_in_regular_use(vehicle)
+                 end
+        save_assessed_value(vehicle, result)
+        result.value
+      end
+
+    private
+
+      def assess_in_regular_use(vehicle, submission_date)
+        net_value = vehicle.value - vehicle.loan_amount_outstanding
+        if too_old_to_count(vehicle, submission_date) || net_value <= vehicle_disregard(submission_date)
+          Result.new(0, false)
+        else
+          Result.new(net_value - vehicle_disregard(submission_date), true)
+        end
+      end
+
+      def assess_not_in_regular_use(vehicle)
+        Result.new(vehicle.value, true)
+      end
+
+      def too_old_to_count(vehicle, submission_date)
+        age_in_months(vehicle, submission_date) >= vehicle_out_of_scope_age(submission_date)
+      end
+
+      # TODO: remove this side effect
+      def save_assessed_value(vehicle, result)
+        vehicle.update!(assessed_value: result.value, included_in_assessment: result.included_in_assessment)
+      end
```

Comment:
I thought the intention was to remove this side effect as well?

Reply:
I believe we left it as optional to include in this ticket or leave for another